boards/pimoroni-servo2040/src/lib.rs

```diff
@@ -0,0 +1,767 @@
+#![no_std]
+
+pub extern crate rp2040_hal as hal;
+
+#[cfg(feature = "rt")]
+extern crate cortex_m_rt;
+#[cfg(feature = "rt")]
+pub use hal::entry;
+
+/// The linker will place this boot block at the start of our program image. We
+/// need this to help the ROM bootloader get our code up and running.
+#[cfg(feature = "boot2")]
+#[link_section = ".boot2"]
+#[no_mangle]
+#[used]
+pub static BOOT2_FIRMWARE: [u8; 256] = rp2040_boot2::BOOT_LOADER_W25Q080;
+
+pub const NUM_SERVOS: u8 = 18;
+pub const NUM_SENSORS: u8 = 6;
+pub const NUM_LEDS: u8 = 6;
+
+pub const SENSOR_1_ADDR: u8 = 0b_0000;
+pub const SENSOR_2_ADDR: u8 = 0b_0001;
+pub const SENSOR_3_ADDR: u8 = 0b_0010;
+pub const SENSOR_4_ADDR: u8 = 0b_0011;
+pub const SENSOR_5_ADDR: u8 = 0b_0100;
+pub const SENSOR_6_ADDR: u8 = 0b_0101;
+pub const VOLTAGE_SENSE_ADDR: u8 = 0b_0110;
+pub const CURRENT_SENSE_ADDR: u8 = 0b_0111;
+
+
+pub const VOLTAGE_GAIN: f32 = 0.28058;
+pub const SHUNT_RESISTOR: f32 = 0.003;
+pub const CURRENT_GAIN: u8 = 69;
+pub const CURRENT_OFFSET: f32 = -0.02;
+
+
+pub use hal::pac;
+
+hal::bsp_pins!(
+    /// GPIO 0 supports following functions:
+    ///
+    /// | Function     | Alias with applied function |
+    /// |--------------|-----------------------------|
+    /// | `SPI0 RX`    | [crate::Servo1Spi0Rx]          |
+    /// | `UART0 TX`   | [crate::Servo1Uart0Tx]         |
+    /// | `I2C0 SDA`   | [crate::Servo1I2C0Sda]         |
+    /// | `PWM0 A`     | [crate::Servo1Pwm0A]           |
+    /// | `PIO0`       | [crate::Servo1Pio0]            |
+    /// | `PIO1`       | [crate::Servo1Pio1]            |
+    Gpio0 {
+        name: servo1,
```

Comment:
All of the pins named servo* are connected to standard servo pinouts (signal, V+, Gnd) and are wired to 180Ohm resistor (see https://cdn.shopify.com/s/files/1/0174/1800/files/servo2040_schematic.pdf?v=1648817752 and https://cdn.shopify.com/s/files/1/0174/1800/files/servo-2040-drawing.png?v=1649088626 for layout along the top). Should I remove the aliases for simplicity since they're unlikely to be used by the other names?

Reply:
The aliases are always optional.
That said, I think 3 of them still make sense for a servo depending on how you're controlling it (PWM/PIO0/PIO1)